evm/src/cross_table_lookup.rs

```diff
@@ -466,16 +469,81 @@ impl<F: Field> CrossTableLookup<F> {
         }
     }
 
-    /// Given a `Table` t and the number of challenges, returns the number of Cross-table lookup polynomials associated to t,
-    /// i.e. the number of looking and looked tables among all CTLs whose columns are taken from t.
-    pub(crate) fn num_ctl_zs(ctls: &[Self], table: Table, num_challenges: usize) -> usize {
+    /// Given a `Table` t and the number of challenges, returns the number of helper and Z polynomials for all
+    /// Cross table lookups associated to t, i.e. the number of polynomials for looking and looked tables
+    /// whose columns are taken from t.
+    pub(crate) fn num_ctl_helper_zs(
+        ctls: &[Self],
+        table: Table,
+        num_challenges: usize,
+        constraint_degree: usize,
+    ) -> usize {
+        let mut num_ctls = 0;
+        for ctl in ctls {
+            let all_tables = std::iter::once(&ctl.looked_table).chain(&ctl.looking_tables);
+            let num_appearances = all_tables.filter(|twc| twc.table == table).count();
+            if num_appearances > 0 {
+                num_ctls += ceil_div_usize(num_appearances, constraint_degree - 1) + 1;
```

Comment:
not sure how impactful it is, but `constraint_degree - 1` being fixed here, we may want to consider dividing only once (same below)?

Reply:
If I understand what you mean correctly, I'm not sure we can do that here: since it's a `ceil_div_usize` for each CTL, we might have an issue if, say, we have 1 appearance in two CTLs: then if we divide at the end, we would get 1, but we should have gotten 2.